modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeCatchAllJavaLongLinesCheck.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.tools.java.parser.JavaParser;
+
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+
+import java.io.File;
+import java.io.IOException;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Nícolas Moura
+ */
+public class UpgradeCatchAllJavaLongLinesCheck extends JavaLongLinesCheck {
+
+	@Override
+	protected String doProcess(
+			String fileName, String absolutePath, String fileContent)
+		throws IOException {
+
+		Matcher matcher = _pattern.matcher(fileName);
+
+		if (!matcher.find() ||
+			!absolutePath.contains("/upgrade/upgrade-catch-all-check") ||
+			!fileName.contains("-before")) {
+
+			return fileContent;
+		}
+
+		setMaxLineLength(80);
+
+		super.doProcess(
+			StringUtil.removeSubstring(fileName, "-before"), absolutePath,
+			fileContent);
+
+		try {
+			return JavaParser.parse(new File(absolutePath), 80);
```

Comment:
2. I remember we disabled JavaParser for Upgrade checks. Do customers want to enable it?

Sorry for these questions, I am not very familiar with Upgrade checks.

Reply:
Hi @ling-alan-huang, do not worry about it. I gave you an overview in Slack.
Please take a look when you can. Thanks!